cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py

```diff
@@ -839,21 +809,35 @@ def launch_default_ingress_controller():
             context['defaultbackend_image'] = \
                 "k8s.gcr.io/defaultbackend-amd64:1.5"
 
+    # Render the default http backend (404) replicationcontroller manifest
+    manifest = addon_path.format('default-http-backend.yaml')
+    render('default-http-backend.yaml', manifest, context)
+    hookenv.log('Creating the default http backend.')
+    try:
+        kubectl('apply', '-f', manifest)
+    except CalledProcessError as e:
+        hookenv.log(e)
+        hookenv.log('Failed to create default-http-backend. Will attempt again next update.')  # noqa
+        hookenv.close_port(80)
+        hookenv.close_port(443)
+        return
+
     # Render the ingress daemon set controller manifest
     context['ssl_chain_completion'] = config.get(
         'ingress-ssl-chain-completion')
     context['ingress_image'] = config.get('nginx-image')
     if context['ingress_image'] == "" or context['ingress_image'] == "auto":
-        images = {'amd64': 'quay.io/kubernetes-ingress-controller/nginx-ingress-controller:0.19.0',  # noqa
-                  'arm64': 'quay.io/kubernetes-ingress-controller/nginx-ingress-controller-arm64:0.19.0',  # noqa
-                  's390x': 'quay.io/kubernetes-ingress-controller/nginx-ingress-controller-s390x:0.19.0',  # noqa
-                  'ppc64el': 'quay.io/kubernetes-ingress-controller/nginx-ingress-controller-ppc64le:0.19.0',  # noqa
+        images = {'amd64': 'quay.io/kubernetes-ingress-controller/nginx-ingress-controller:0.16.1',  # noqa
+                  'arm64': 'quay.io/kubernetes-ingress-controller/nginx-ingress-controller-arm64:0.16.1',  # noqa
```

Comment:
I think this ingress controller will not work for arm. Have a look at this comment: https://github.com/kubernetes/ingress-nginx/pull/3214#pullrequestreview-163161031 pointing to this bug https://github.com/kubernetes/ingress-nginx/issues/2802 . On microk8s+arm64 I had to revert all the way to  tag "0.11.0"

Reply:
Good to know, thanks. I think we can safely ignore that concern in this PR though.